src/gui/screen_menu_calibration.c

```diff
@@ -78,13 +86,25 @@ int screen_menu_calibration_event(screen_t *screen, window_t *window, uint8_t ev
             wizard_run_complete();
             break;
         case MI_AUTO_HOME:
+            marlin_event_clr(MARLIN_EVT_CommandBegin);
             marlin_gcode("G28");
-            gui_dlg_wait(gui_marlin_busy_callback);
+            while (!marlin_event_clr(MARLIN_EVT_CommandBegin))
+                marlin_client_loop();
+            gui_dlg_wait(gui_marlin_G28_or_G29_in_progress);
             break;
         case MI_MESH_BED:
-            marlin_gcode("G28");
+            if (!marlin_all_axes_homed()) {
+                marlin_event_clr(MARLIN_EVT_CommandBegin);
+                marlin_gcode("G28");
+                while (!marlin_event_clr(MARLIN_EVT_CommandBegin))
```

Comment:
does marlin_event_clr() switch RTOS tasks, so that the CPU can do something else while waiting here (burning CPU cycles for nothing)?